src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,44 @@
 package core.mate.academy.service;
 
+import core.mate.academy.machine.impl.BulldozerProducerImpl;
+import core.mate.academy.machine.impl.ExcavatorProducerImpl;
+import core.mate.academy.machine.impl.TruckProducerImpl;
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<Machine> machineList = new ArrayList<>();
```

Comment:
Consider creating a local variable in your implementation of `getAll(Class type)` method to hold the result from the producer's get method before returning it.